File: algorithm/src/main/java/baekjoon/p09946/Main.java
package baekjoon.p09946;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int i = 1;

        while (true) {
            String a = sc.next();
            String b = sc.next();

            if (a.equals("END")) {
                break;
            }

            System.out.println(String.format("Case %d: %s", i++, solve(a, b)));
        }
    }

    private static String solve(String a, String b) {
        int[] cntA = cnt(a);
        int[] cntB = cnt(b);

        for (int i = 0; i < cntA.length; i++) {
            if (cntA[i] != cntB[i]) {
                return "different";
            }
        }

        return "same";
    }

    private static int[] cnt(String str) {
        int[] ret = new int[26];

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);
            ret[ch - 'a']++;
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p09946/Main.kt
package baekjoon.p09946

fun main(args: Array<String>) {
    var i = 1
    while (true) {
        val a = readLine()!!
        val b = readLine()!!

        if (a == "END") {
            break
        }

        println("Case ${i++}: ${solve(a, b)}")
    }
}

private fun solve(a: String, b: String): String {
    val cntA = ('a'..'z').map { a.count { ch -> ch == it } }
    val cntB = ('a'..'z').map { b.count { ch -> ch == it } }

    return when (cntA == cntB) {
        true -> "same"
        false -> "different"
    }
}